agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3988,9 +3955,10 @@ bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMess
         return false;
     }
 
-    if (intel_agent) {
+    auto beerocks_header = message_com::parse_intel_vs_message(cmdu_rx);
+    if (beerocks_header) {
         LOG(INFO) << "Intel controller join response";
-        if (!parse_intel_join_response(sd, cmdu_rx)) {
+        if (!parse_intel_join_response(sd, *beerocks_header)) {
```

Comment:
It would have been nice to have done this change in a separate, preparative commit (removing intel_agent and relying on `if (beerocks_header)` is something that can be done independent of the change to `parse(true)`).

nitpick, of course.

Reply:
Splitting now is too much extra effort but I agree, will take it into account next time.